src/BPINN_ode.jl

```diff
@@ -99,48 +99,38 @@ struct BNNODE{C, K, ST <: Union{Nothing, AbstractTrainingStrategy}, IT, A, M,
     l2std::Vector{Float64}
     phystd::Vector{Float64}
     dataset::D
-    init_params::I
     physdt::Float64
+    n_leapfrog::Int64
+    NUTSkwargs::N
+    HMCDAkwargs::H
     nchains::Int64
-    autodiff::Bool
+    init_params::I
     Integrator::IT
     Adaptor::A
-    targetacceptancerate::Float64
     Metric::M
+    targetacceptancerate::Float64
     jitter_rate::Float64
     tempering_rate::Float64
-    max_depth::Int64
-    Δ_max::Int64
-    n_leapfrog::Int64
-    δ::Float64
-    λ::Float64
+    autodiff::Bool
     progress::Bool
     verbose::Bool
-
-    function BNNODE(chain, Kernel = HMC; strategy = nothing,
-        draw_samples = 2000,
-        priorsNNw = (0.0, 2.0), param = [nothing], l2std = [0.05],
-        phystd = [0.05], dataset = [nothing],
-        init_params = nothing,
-        physdt = 1 / 20.0, nchains = 1,
-        autodiff = false, Integrator = Leapfrog,
-        Adaptor = StanHMCAdaptor, targetacceptancerate = 0.8,
-        Metric = DiagEuclideanMetric, jitter_rate = 3.0,
-        tempering_rate = 3.0, max_depth = 10, Δ_max = 1000,
-        n_leapfrog = 20, δ = 0.65, λ = 0.3, progress = false,
-        verbose = false)
-        new{typeof(chain), typeof(Kernel), typeof(strategy), typeof(Integrator),
-            typeof(Adaptor),
-            typeof(Metric), typeof(init_params), typeof(param),
-            typeof(dataset)}(chain, Kernel, strategy, draw_samples,
-            priorsNNw, param, l2std,
-            phystd, dataset, init_params,
-            physdt, nchains, autodiff, Integrator,
-            Adaptor, targetacceptancerate,
-            Metric, jitter_rate, tempering_rate,
-            max_depth, Δ_max, n_leapfrog,
-            δ, λ, progress, verbose)
-    end
+end
+function BNNODE(chain, Kernel = HMC; strategy = nothing,
+    draw_samples = 2000, priorsNNw = (0.0, 2.0), param = nothing,
+    l2std = [0.05], phystd = [0.05], dataset = [nothing], physdt = 1 / 20.0,
+    n_leapfrog = 20, NUTSkwargs = (δ = 0.65, max_depth = 10, Δ_max = 1000),
```

Comment:
Either one of NUTSkwards or HMCDAkwargs will be used at a time right? Make this a single arg `mcmc_kwargs` or something like that and then depending on the typeof the mcmc method you have you can use them

Reply:
fixed, even the lower level call is more streamlined now